### ISSUE TITLE
[BUG] node disk manager relies on unstable devPath to retrieve fiilesystem info

### ISSUE THREAD

**Issue by weihanglo** [harvester/harvester#1829]
**Describe the bug**
node disk manager relies on unstable **dev path** to retrieve fiilesystem info, which is not always consistent between each boot.

**To Reproduce**
1. Read the source code: https://github.com/harvester/node-disk-manager/blob/73948d5a0d2a0f317709d49f689791325d3654f7/pkg/block/block_device.go#L95-L104

**Expected behavior**
Depends on a stable and unique device naming mechanism such like filesystem UUID

**Support bundle**

**Environment:**
 - Harvester ISO version: v1.0.0
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**

**Comment by weihanglo** [comment]
## Test Plan

1. Prepare a harvester cluster (single node is sufficient)
2. Provision two extra disks onto the host
3. Shutdown the host and then swap those two disks (either physical or virtual). 
   - This not always leads to devPath swapping but just give it a try.
   - If your test environment is libvirt/qemu, swap the filename and WWN.
5. Reboot the host. 
6. Expect that devPaths of those disks are swapped, but the disks on "Host" page are healthy and stable regardless underlying devPaths changes.

**Comment by TachunLin** [comment]
Verified **not fixed** on `master-b09e3233-head` (02/14). Move this issue to implement pipeline.

### Result
After swapping the source path and the unique <wwn> value content of the two extra disk and power on harvester machine.  

* extra disk 1 display failed to mount error and extra disk 2 display create mount point does not match the specific path error
![image](https://user-images.githubusercontent.com/29251855/153828113-67a77caf-1070-4a7f-ae22-0e0539417daf.png)

* Disk stage display `Warning` status
![image](https://user-images.githubusercontent.com/29251855/153828023-4c67f31f-127a-4e94-ac5c-4edb5189e60f.png)
 


### Test Information
* Test Environment: 1 node harvester on local kvm machine
* Harvester version: `master-b09e3233-head` (02/14)


### Verify Steps
1. Create a virtual machine
2. Add two extra SCSI bus disks
3. Edit the XML content of the extra disk 1 (130GB)
4. Add the `<wwn>1000000000000005</wwn>` unique value to the disk 
5. Edit the XML content of the extra disk 2 (100GB)
6. Add the `<wwn>1000000000000006</wwn>` unique value to the disk
7. ISO install harvester on non-extra disk
![image](https://user-images.githubusercontent.com/29251855/153813484-a3dec175-530b-4004-9baf-69310b68d606.png)
8. After harvester node installation complete, attach the two extra disks on the host page 
![image](https://user-images.githubusercontent.com/29251855/153815882-de5a205f-f756-47d2-b936-a01841e6bbdc.png)

![image](https://user-images.githubusercontent.com/29251855/153816506-6a5745fc-1668-4289-a6d8-139dfac208b8.png)
9. Shutdown harvester node 
10. Edit the disk 1 xML content in Virt manager 
11. Swap the `<source> path` and `<wwn>` content between disk 1 and disk 2 
12. Power on harvester node machine
13. Check disk status in the host page 

### Additional Context

* Extra disk 1 XML configuration after swapped
```
<disk type="file" device="disk">
  <driver name="qemu" type="qcow2"/>
  <source file="/var/lib/libvirt/images/harvester-one-220214-3.qcow2" index="2"/>
  <backingStore/>
  <target dev="sda" bus="scsi"/>
  <wwn>1000000000000006</wwn>
  <alias name="scsi0-0-0"/>
  <address type="drive" controller="0" bus="0" target="0" unit="0"/>
</disk>
```

* Extra disk 2 XML configuration after swapped
```
<disk type="file" device="disk">
  <driver name="qemu" type="qcow2"/>
  <source file="/var/lib/libvirt/images/harvester-one-220214-2.qcow2" index="1"/>
  <backingStore/>
  <target dev="sdb" bus="scsi"/>
  <wwn>1000000000000005</wwn>
  <alias name="scsi0-0-1"/>
  <address type="drive" controller="0" bus="0" target="0" unit="1"/>
</disk>
```

* disk allocation status
```
harveser-one-220214:~ # lsblk
NAME   MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
loop0    7:0    0  3.2G  1 loop /
sda      8:0    0  130G  0 disk 
└─sda1   8:1    0  130G  0 part /var/lib/harvester/extra-disks/97c925ae62d282b61271c8c571c63326
sdb      8:16   0  100G  0 disk 
└─sdb1   8:17   0  100G  0 part 
sdc      8:32   0  200G  0 disk 
├─sdc1   8:33   0  960K  0 part 
├─sdc2   8:34   0   50M  0 part /oem
├─sdc3   8:35   0   15G  0 part /run/initramfs/cos-state
├─sdc4   8:36   0    8G  0 part 
├─sdc5   8:37   0   60G  0 part /usr/local
└─sdc6   8:38   0  117G  0 part /var/lib/harvester/defaultdisk
sdd      8:48   0   50G  0 disk /var/lib/kubelet/pods/ce6f7828-1e58-451a-a91b-8ee74666ff37/volume-subpaths/pvc-ea4a0a58-e4a5-4725-a4ce-b783651b1
sde      8:64   0   10M  0 disk /var/lib/kubelet/pods/a7bae4f1-91ab-43b3-8e8a-3f313f5f6225/volumes/kubernetes.io~csi/pvc-4924ded4-6759-47ef-8e69
sr0     11:0    1 1024M  0 rom
```

**Comment by TachunLin** [comment]
Verified **fixed** on master-bfd6a8a9-head. (02/27). Close this issue.

### Result
After swapping, devPaths of extra disk are also swapped and the status of disks on "Host" page are healthy and stable

![image](https://user-images.githubusercontent.com/29251855/156134016-2b2a9bf9-a230-49ad-9ba2-db93cab296d2.png)


* Before swap 
  ```
  rancher@harvester-one-220227:~> lsblk
  NAME   MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
  loop0    7:0    0  3.2G  1 loop /
  sda      8:0    0  100G  0 disk /var/lib/harvester/extra-disks/42913fd260cf49e87679a1b976384bff
  sdb      8:16   0  130G  0 disk /var/lib/harvester/extra-disks/7e879b8776022bcb6e1b833bdf00cc40
  sdc      8:32   0  200G  0 disk 
  ├─sdc1   8:33   0  960K  0 part 
  ├─sdc2   8:34   0   50M  0 part /oem
  ├─sdc3   8:35   0   15G  0 part /run/initramfs/cos-state
  ├─sdc4   8:36   0    8G  0 part 
  ├─sdc5   8:37   0   60G  0 part /usr/local
  └─sdc6   8:38   0  117G  0 part /var/lib/harvester/defaultdisk
  sdd      8:48   0   10M  0 disk /var/lib/kubelet/pods/330493cb-22c8-4424-8ea2-bd7572158200/volumes/kubernetes.io~csi/pvc-9998b18f-6311-4fc6-8b8f
  sde      8:64   0   50G  0 disk /var/lib/kubelet/pods/0bc09714-ace2-4a31-a7ef-9d75f51f4cc1/volume-subpaths/pvc-8a8d8a3e-dc04-4fa5-b632-9f95cbdab
  sr0     11:0    1 1024M  0 rom
  ```

* After swap
  ```
  rancher@harvester-one-220227:~> lsblk
  NAME   MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
  loop0    7:0    0  3.2G  1 loop /
  sda      8:0    0  130G  0 disk /var/lib/harvester/extra-disks/7e879b8776022bcb6e1b833bdf00cc40
  sdb      8:16   0  100G  0 disk /var/lib/harvester/extra-disks/42913fd260cf49e87679a1b976384bff
  sdc      8:32   0  200G  0 disk 
  ├─sdc1   8:33   0  960K  0 part 
  ├─sdc2   8:34   0   50M  0 part /oem
  ├─sdc3   8:35   0   15G  0 part /run/initramfs/cos-state
  ├─sdc4   8:36   0    8G  0 part 
  ├─sdc5   8:37   0   60G  0 part /usr/local
  └─sdc6   8:38   0  117G  0 part /var/lib/harvester/defaultdisk
  sdd      8:48   0   10M  0 disk /var/lib/kubelet/pods/d523266c-abff-45c3-b0db-603cf099a32b/volumes/kubernetes.io~csi/pvc-9998b18f-6311-4fc6-8b8f
  sde      8:64   0   50G  0 disk /var/lib/kubelet/pods/bcd1ac36-1103-440e-a588-5cf17bdf0083/volume-subpaths/pvc-8a8d8a3e-dc04-4fa5-b632-9f95cbdab
  sr0     11:0    1 1024M  0 rom
  ```

### Test Information
* Test Environment: 1 node harvester on local kvm machine
* Harvester version: master-bfd6a8a9-head


### Verify Steps
Same as https://github.com/harvester/harvester/issues/1829#issuecomment-1038810488

### Additional Context
Similar related issue #1874
